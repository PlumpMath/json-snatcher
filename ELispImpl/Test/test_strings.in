"Wee this is a string"
"abcde 12345 6789 10"
"abcde" "12345" "6789" "10"
"abcde\"" 
"12345\\" 
"sterling\\\\"
