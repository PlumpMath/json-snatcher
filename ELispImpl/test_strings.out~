"abcde 12345 6789 10"
"abcde"
"12345"
"6789"
"10"
    def test_strings_w_quote_escapes(self):
        test_str = r'"abcde\"" "12345\\" "sterling\\\\" "\\6789\\\" \"10\uabcd2\q0" "\\\\"""'
        result = self.regex.findall(test_str)
"abcde\""
"12345\\"
"sterling\\\\"
"\\6789\\\" \"10\uabcd2\q0"
"\\\\"
""

    def test_reg_escapes(self):
        escapes = [r'\"', r'\\', r'\/', r'\b', r'\f', r'\n', r'\r', r'\t', r'\uAa23']

        for escape in escapes:
            test_str = r'"abcde 12345 ' + escape + '6789 10"'
            result = self.regex.findall(test_str)
            self.assertEqual(r'"abcde 12345 ' + escape + '6789 10"', result[0])

    def test_escapes_dbl(self):
        escapes = [r'\"', r'\\', r'\/', r'\b', r'\f', r'\n', r'\r', r'\t', r'\uAa23']

        for escape in escapes:
            test_str = r'"abcde 12345 \\' + escape + r'6789 10"'
            result = self.regex.findall(test_str)
            self.assertEqual(r'"abcde 12345 \\' + escape + r'6789 10"', result[0])

    def test_mt_string(self):
        test_str = r'""'
        result = self.regex.findall(test_str)
        self.assertEqual(r'""', result[0])

    def test_url(self):
        test_str = r'"http://www.example.com/image/481989943"'
        result = self.regex.findall(test_str)
        self.assertEqual(r'"http://www.example.com/image/481989943"', result[0])
        
    def test_rfc_object(self):

        expected = [
            '"Image"',
            '"Width"',
            '"Height"',
            '"Title"', '"View from 15th Floor"',
            '"Thumbnail"',
            '"Url"','"http://www.example.com/image/481989943"',
            '"Height"',
            '"Width"', '"100"',
            '"IDs"']

        with open('rfc_object.json','rb') as f:
            test_str = f.read()
            test_str = r'' + test_str

            result = self.regex.findall(test_str)
            i = 0
            for r in result:
                self.assertEqual(r, expected[i])
                i += 1

    def test_rfc_array(self):
        expected = [
         '"precision"', '"zip"',
         '"Latitude"',
         '"Longitude"',
         '"Address"', '""',
         '"City"', '"SAN FRANCISCO"',
         '"State"', '"CA"',
         '"Zip"', '"94107"',
         '"Country"', '"US"',
         '"precision"', '"zip"',
         '"Latitude"',
         '"Longitude"',
         '"Address"', '""',
         '"City"', '"SUNNYVALE"',
         '"State"', '"CA"',
         '"Zip"', '"94085"',
         '"Country"', '"US"']

        with open('rfc_array.json','rb') as f:
            test_str = f.read()
            test_str = r'' + test_str

            result = self.regex.findall(test_str)
            i = 0
            for r in result:
                self.assertEqual(r, expected[i])
                i += 1

    def test_long_json(self):
        with open('aws.json','rb') as f:
            test_str = f.read()
            test_str = r'' + test_str
            result = self.regex.findall(test_str)
            self.assertEqual(result[0],'"config"')
            self.assertEqual(result[10],'"us-east"')
            self.assertEqual(result[len(result)-1], '"vers"')
