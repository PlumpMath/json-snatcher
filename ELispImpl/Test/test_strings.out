Regular Tests:
"abcde 12345 6789 10"
"abcde"
"12345"
"6789"
"10"
"abcde\""
"12345\\"
"sterling\\\\"
"\\6789\\\" \"10\uabcd2\q0"
"\\\\"
""
""
"http://www.example.com/image/481989943"

RFC Object Example:
"Image"
"Width"
"Height"
"Title"
"View from 15th Floor"
"Thumbnail"
"Url"
"http://www.example.com/image/481989943"
"Height"
"Width"
"IDs"
RFC Example Array:
"precision"'
"zip"
"Latitude"
"Longitude"
"Address"
""
"City"
"SAN FRANCISCO"
"State"
"CA"
"Zip"
"94107"
"Country"
"US"
"precision"
"zip"
"Latitude"
"Longitude"
"Address"
""
"City"
"SUNNYVALE"
"State"
"CA"
"Zip"
"94085"
"Country"
"US"
Long JSON:

            self.assertEqual(result[0],'"config"')
            self.assertEqual(result[10],'"us-east"')
            self.assertEqual(result[len(result)-1], '"vers"')
